app/views/shared/_one_time_code_input.html.erb

```diff
@@ -0,0 +1,28 @@
+<%#
+Renders a numeric OTP code input field. In addition to the locals described below, all additional
+local assigns will be applied directly to the text input as a hash of HTML attributes.
+
+locals:
+* name: Field name. Defaults to `:code`.
+* value: Field value. Defaults to `''`.
+%>
+<%
+name = local_assigns[:name] || :code
+classes = ['field monospace usa-input one-time-code-input']
+classes << local_assigns[:class] if local_assigns[:class]
```

Comment:
given the `**local_assigns` at the end, I think we should `.delete` keys as we pull them out to make sure we don't collide accidentally:

```suggestion
name = local_assigns.delete(:name) || :code
classes = ['field monospace usa-input one-time-code-input']
classes << local_assigns.delete(:class) if local_assigns[:class]
```

Reply:
I did consider this possibility, and for the two values here, it couldn't actually meaningfully collide in that...

1. If `name` is assigned from `local_assigns`, this is meaningful and at-worst redundant with what `text_field_tag` is already using it for ([source](https://github.com/rails/rails/blob/2bbb519482dbc7969321719cb515216cf1515bfc/actionview/lib/action_view/helpers/form_tag_helper.rb#L235)).
2. The placement of `class: classes` after `**local_assigns` is intended to re-override any `class` value from `local_assigns`.

That being said, I can see how your suggestion would at least make this intention clearer, and set a precedent for future options.

I'd actually been trying to find some examples earlier of an equivalent to something like `const { a, b, ...rest } = obj;` in JavaScript, but for Ruby.

I guess that would be something like...

```rb
a = obj.delete(:a)
b = obj.delete(:b)
rest = obj
```

Reply:
I thought Ruby would blow up if splat args collided with a literal value but it doesn't :yikes:

```ruby
def foobar(a:)
  a
end
x = { a: 1 }
foobar(a: 2, **x)
=> 1
foobar(**x, a: 2)
=> 2
```

Reply:
My mental model of it had been to see the splat args as effectively equating these two:

```rb
{ type: 'text', inputmode: 'numeric', **local_assigns, class: classes }
```

```rb
{ type: 'text', inputmode: 'numeric' }.merge(local_assigns).merge(class: classes)
```

Reply:
Debating also if it might be better to create something like an `input_html` local, a hash of HTML attributes that we don't mess with. There's some precedent for this, but given how we handle `name` and `class`, I think it could be confusing no matter which way we go about it, since if we were to considering them "top-level" locals, it could be confusing if someone considers them as HTML attributes. And if we expect from in `input_html`, we'd still have to modify them (`:class`) or use them in ways where they aren't just directly applied as HTML attributes (`:name`).